File: programmers/python/입문100/13.1_컨트롤제트/13-1.py
# 1. 반복문

def solution(string):

    str_list = list(string.split())[::-1]
    N = len(str_list)

    idx = 0
    num_list = []

    while idx < N : # 끝까지 갈때까지 진행

        if str_list[idx] != 'Z':
            num_list.append(str_list[idx])
            idx += 1
        else :
            idx += 2

    num_list = [int(i) for i in num_list]
    return sum(num_list)

# 2. 스택

def solution(string):
    stack = []
    for s in string.split():
        if s == 'Z' and stack:
            stack.pop()
        else:
            stack.append(int(s))
    return sum(stack)

File: programmers/java/기초124/13.1_n번째원소부터/13_1.java
class Solution {
    public int[] solution(int[] num_list, int n) {
        // int[] answer = {};
        int len = num_list.length;
        int[] answer = new int[len - (n - 1)];

        for (int i = n-1; i < len; i++) {
            answer[i - (n-1)] = num_list[i];
        }

        return answer;
    }
}